src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.cs

```diff
@@ -33,9 +33,10 @@ namespace Microsoft.Python.Analysis.Analyzer.Evaluation {
     /// Helper class that provides methods for looking up variables
     /// and types in a chain of scopes during analysis.
     /// </summary>
-    internal sealed partial class ExpressionEval : IExpressionEvaluator {
```

Comment:
I think dropping `IExpressionEvaluator` means `ExpressionEval.Hints.cs` is the file which declares that this class implements the interface, which seems a bit strange (considering I think the interface is really implemented in this file).

Reply:
It can be in any file, but yes, it is better to have it in the main one.